apps/worker/src/app/workflow/usecases/run-job/run-job.usecase.ts

```diff
@@ -114,19 +119,54 @@ export class RunJob {
     }
   }
 
+  /*
+   * If the following condition is met,
+   * - transactions were merged to the main delayed digest.
+   * - the main delayed digest was canceled.
+   * that mean that we need to assign a new active digest follower job to replace it.
+   * so from now on we will continue the follower transaction as main digest job.
+   */
+  private assignNewDigestExecutor(activeDigestFollower: JobEntity): JobEntity {
+    return activeDigestFollower;
+  }
+
+  private isCanceledMainDigest(type: StepTypeEnum | undefined, status: JobStatusEnum) {
+    return type === StepTypeEnum.DIGEST && status === JobStatusEnum.CANCELED;
+  }
+
   @Instrument()
-  private async delayedEventIsCanceled(job: JobEntity): Promise<boolean> {
+  private async delayedEventIsCanceled(
+    job: JobEntity
+  ): Promise<{ canceled: boolean; activeDigestFollower: JobEntity | null }> {
+    let activeDigestFollower: JobEntity | null = null;
+
     if (job.type !== StepTypeEnum.DIGEST && job.type !== StepTypeEnum.DELAY) {
-      return false;
+      return { canceled: false, activeDigestFollower };
     }
 
-    const count = await this.jobRepository.count({
```

Comment:
Removed this query because we already have the newly fetched job and its status.